amgut/lib/data_access/sql_connection.py

```diff
@@ -1,72 +1,204 @@
-from __future__ import division
-
 # -----------------------------------------------------------------------------
-# Copyright (c) 2014--, The American Gut Development Team.
+# Copyright (c) 2014--, The Qiita Development Team.
 #
 # Distributed under the terms of the BSD 3-clause License.
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
-
+from __future__ import division
 from contextlib import contextmanager
-from psycopg2 import connect, Error as PostgresError
+from itertools import chain
+from functools import partial, wraps
+from datetime import date, time, datetime
+
+from psycopg2 import (connect, ProgrammingError, Error as PostgresError,
+                      OperationalError)
 from psycopg2.extras import DictCursor
+from psycopg2.extensions import (
+    ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_READ_COMMITTED,
+    TRANSACTION_STATUS_IDLE)
 
 from amgut.lib.config_manager import AMGUT_CONFIG
 
 
 class SQLConnectionHandler(object):
```

Comment:
Not yet, as the Survey object relies on it and I don't want to do edits on that without tests in place.